services/github/github-pull-request-status.service.js

```diff
@@ -1,85 +0,0 @@
-'use strict'
-
-const countBy = require('lodash.countby')
-const LegacyService = require('../legacy-service')
-const {
-  makeColorB,
-  makeBadgeData: getBadgeData,
-  makeLogo: getLogo,
-} = require('../../lib/badge-data')
-const {
-  checkStateColor: githubCheckStateColor,
-  checkErrorResponse: githubCheckErrorResponse,
-} = require('./github-helpers')
-
-module.exports = class GithubPullRequestStatus extends LegacyService {
-  static registerLegacyRouteHandler({ camp, cache, githubApiProvider }) {
-    camp.route(
-      /^\/github\/status\/(s|contexts)\/pulls\/([^/]+)\/([^/]+)\/(\d+)\.(svg|png|gif|jpg|json)$/,
-      cache((queryParams, match, sendBadge, request) => {
-        const [, which, owner, repo, number, format] = match
-        const issueUri = `/repos/${owner}/${repo}/pulls/${number}`
-        const badgeData = getBadgeData('checks', queryParams)
-        if (badgeData.template === 'social') {
-          badgeData.logo = getLogo('github', queryParams)
```

Comment:
We've lost some functionality here:
![](https://shields-staging-pr-2253.herokuapp.com/github/status/s/pulls/badges/shields/1110.svg?style=social) - https://shields-staging-pr-2253.herokuapp.com/github/status/s/pulls/badges/shields/1110.svg?style=social
![](https://img.shields.io/github/status/s/pulls/badges/shields/1110.svg?style=social) - https://img.shields.io/github/status/s/pulls/badges/shields/1110.svg?style=social

This seems like a good thing to implement in a base class which the GH service badges inherit from

Reply:
Ah, yes, thanks for catching that!

This "default social logo" functionality belongs in BaseService. I think I'd probably call it `logo` or `socialLogo` and put it in `defaultBadgeData`. An option like this could also facilitate adding a `?logo=yes`-type option in the future.

The base Github service doesn't seem like a bad place to specify it, though many subclasses will need to override it. It might be less error-prone to include it in each badge class. That way it'll never be forgotten if a one of the subclasses gets a default badge data property at a later date.

Reply:
Ha! Don't you just love realizing features like these are already implemented?

https://github.com/badges/shields/blob/bc4bd79e90058426278a973e7f742761b90d3b9c/services/base.js#L314-L317